### DISCUSSION TITLE
Can't connect to MariaDB using Default Portainer Stack

### DISCUSSION THREAD

**Discussion by Compr0mzd** [firefly-iii/firefly-iii#8984] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Using the portainer stack given in the documentation here [https://docs.firefly-iii.org/how-to/data-importer/installation/portainer/](url). I did setup everything inside the documentation as precised and when the lauching the stack I get these errors : 
`Could not poll the database: SQLSTATE[HY000] [1130] Host '172.19.0.4' is not allowed to connect to this MariaDB server (Connection: mysql, SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) ` on the webpage
Inside the DB container I get these logs : 
```
2024-06-18 13:56:28 35 [Warning] Aborted connection 35 to db: 'unconnected' user: 'unauthenticated' host: '172.19.0.4' (This connection closed normally without authentication)

2024-06-18 13:56:28 36 [Warning] Aborted connection 36 to db: 'unconnected' user: 'unauthenticated' host: '172.19.0.4' (This connection closed normally without authentication)
```

I'm using a Portainer Instance installed on my UGREEN NAS locally, 
I tried adding the ports inside the docker compose file : 
```
ports:
      - '3306:3306'

```

It did not help.

### Debug information

N/A

**Comment by JC5** [comment]
> 2024-06-18 13:56:28 35 [Warning] Aborted connection 35 to db: 'unconnected' user: 'unauthenticated' host: '172.19.0.4' (This connection closed normally without authentication)

This should be unrelated. It is used by a script that checks if the database is even running, you can see this in the Firefly III logs.

I'll check this later today on my own portainer.

**Comment by Protonix-909** [comment]
I fixed this issue by using the comment from this Reddit post. [https://www.reddit.com/r/docker/comments/xe37gg/comment/k113c0d/?utm_source=share&utm_medium=web2x&context=3](url)

Use command below for logging into database server.

docker exec -it firefly3-db-1 mariadb --user root -pPassword123

Once in view your user table with the command below.

SELECT user, host FROM mysql.user;

If you don't see the matching user with the IP of your app server you will need to add it(View logs on database server to find the username and IP that is failing). Command below to add user account.

CREATE USER 'firefly'@'172.25.0.2' IDENTIFIED BY 'Password123';

GRANT ALL PRIVILEGES ON *.* TO 'firefly'@'[172.25.0.2](https://172.25.0.2/)';

Finally Flush Privileges to apply changes.

FLUSH PRIVILEGES;

Hopefully this helps fix your problem.

**Reply by Compr0mzd** [reply]
I deleted the RANDOM_PASSWORD env and added the MYSQL_ROOT_PASSWORD env varialbe and got the error below
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)

**Comment by JC5** [comment]
So, did you both not use the stack.env file?

**Reply by Compr0mzd** [reply]
I did use the stack.env file

**Reply by Compr0mzd** [reply]
However now it works

**Comment by JC5** [comment]
Works as expected for me. Perhaps it's PEBCAK?

**Comment by Compr0mzd** [comment]
I just deleted the MYSQL_RANDOM_ROOT_PASSWORD=yes and replaced with my own MYSQL_ROOT_PASSWORD and it stopped refusing connection
